### ISSUE TITLE
Plugin crashes Jellyfin when trying to sync favourited track that has no metadata

### ISSUE THREAD

**Issue by physicalmemory** [jesseward/jellyfin-plugin-lastfm#57]
Hi there,

Thanks for your work in developing this plugin. I've encountered a bug that looks to be because I favourited a track that has no metadata in Jellyfin, which the plugin then tries to sync as a loved track to last.fm. The resulting error:

```
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Jellyfin.Plugin.Lastfm.Api.LastfmApiClient.LoveTrack(Audio item, LastfmUser user, Boolean love)
   at Jellyfin.Plugin.Lastfm.ServerEntryPoint.UserDataSaved(Object sender, UserDataSaveEventArgs e)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
```

Then crashes my Jellyfin instance entirely. There is functionality included to skip scrobbling a track when no artist or track metadata is detected, could that potentially also be applied to loved tracks? That may work to resolve this issue.

**Comment by jesseward** [comment]
nice one, thanks for reporting @physicalmemory. I was able to repro this as well. I will work on a fix for this.

```sh
[15:54:06] [FTL] [34] Main: Unhandled Exception
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Jellyfin.Plugin.Lastfm.Api.LastfmApiClient.LoveTrack(Audio item, LastfmUser user, Boolean love) in /workspaces/jellyfin-plugin-lastfm/Jellyfin.Plugin.Lastfm/Api/LastfmApiClient.cs:line 142
   at Jellyfin.Plugin.Lastfm.ServerEntryPoint.UserDataSaved(Object sender, UserDataSaveEventArgs e) in /workspaces/jellyfin-plugin-lastfm/Jellyfin.Plugin.Lastfm/ServerEntryPoint.cs:line 98
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
Unhandled exception. System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Jellyfin.Plugin.Lastfm.Api.LastfmApiClient.LoveTrack(Audio item, LastfmUser user, Boolean love) in /workspaces/jellyfin-plugin-lastfm/Jellyfin.Plugin.Lastfm/Api/LastfmApiClient.cs:line 142
   at Jellyfin.Plugin.Lastfm.ServerEntryPoint.UserDataSaved(Object sender, UserDataSaveEventArgs e) in /workspaces/jellyfin-plugin-lastfm/Jellyfin.Plugin.Lastfm/ServerEntryPoint.cs:line 98
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
```

**Comment by jesseward** [comment]
hey @physicalmemory, this has been addressed in `v8.0.0.4` of the plugin. It's been pushed to the Plugin Repo as well (https://jellyfin-repo.jesseward.com/manifest.json).

You can uninstall and then reinstall the plugin to pick up the new version. Thanks again for reporting
